index.ts

```diff
@@ -21,58 +23,49 @@ enum StartMode {
     waitFirst = 'waitFirst'
 }
 
-enum IntervalMode {
-    fixed = 'fixed',
-    afterFinish = 'afterFinish'
-}
-
 class Repeater {
     action: () => Promise<any>;
-    logger: loggerInterface;
-    limit: number;
+    logger: LoggerInterface;
+    limit!: number;
     runs: number;
-    intervalId: number;
 
-    static sleep:(interval: number) => Promise<any>;
+    static sleep: (interval: number) => Promise<any>;
 
-    static defaultLogger: loggerInterface = console;
-    static voidLogger: loggerInterface = new VoidLogger();
+    static defaultLogger: LoggerInterface = console;
+    static voidLogger: LoggerInterface = new VoidLogger();
     static startMode = StartMode;
-    static intervalMode = IntervalMode;
 
-    constructor(action: () => Promise<any>, options: {logger?:loggerInterface}={}) {
+    constructor(action: () => Promise<any>, options: { logger?: LoggerInterface } = {}) {
         this.action = action;
         this.logger = options.logger ?? Repeater.defaultLogger;
         this.runs = 0;
     }
+
     _formatRepeatLimit(limit: number | null): string {
         return limit === null ? 'forever' : `${limit} times`;
     }
-
-    _executeSleep(interval) {
-        this.logger.debug(`sleeping for ${interval} ms...`);
-        return Repeater.sleep(interval);
-    }
-
-    async continuous(interval: number,limit: number | null, 
-        options: {startMode?: StartMode, intervalMode?: IntervalMode} = {
-            startMode: StartMode.actionFirst,
-            intervalMode: IntervalMode.afterFinish
-        }
-    ): recursivePromise | Promise<void | null> {
+    
+    async continuous(
+        interval: number, 
+        limit: number | null, 
+        options: { startMode?: StartMode } = {}

```

Comment:
lost the defaults?

Reply:
https://github.com/albertdiones/add_repeater/commit/99dc4d0ee336660f7a2628596ded91bf0cc15d27

Reply:
https://github.com/albertdiones/add_repeater/blob/27abe0f0cfb0eff8c5a4ad586eac5bee2c2fae69/index.ts